### ISSUE TITLE
Topic required on KafkaProducerMessage duplicated in CanProduceMessages

### ISSUE THREAD

**Issue by thetomcake** [mateusjunges/laravel-kafka#15]
Hi Guys,

From the [documentation here](https://github.com/mateusjunges/laravel-kafka#configure-the-message-body) I constructed the following code:

```php
use Junges\Kafka\Facades\Kafka;
use Junges\Kafka\Message\Message;

$topic = 'testtopic';
$headers = ['example' => 'header'];
$body = ['example' => 'body'];

Kafka::publishOn(config('kafka.brokers'), $topic)
    ->withMessage(new Message(headers: $headers, body: $body))
    ->send();
```

This errors with `TypeError: Junges/Kafka/AbstractMessage::__construct(): Argument #1 ($topicName) not passed` as $topicName is a required constructor parameter on `\Junges\Kafka\AbstractMessage`

I can just add the topicName constructor parameter but it seems a little weird given that the topic is defined in the `publishOn` method in the previous line. Looking at the implementation of the `\Junges\Kafka\Contracts\CanProduceMessages` it does not seem to actually use the topicName on the Message interface (for the producers anyway).

Can you clarify:
1. If I am misunderstanding this library entirely
2. If the documentation is incorrect
3. What should be happening here.

Happy to assist with PRs but wanted to ensure I understood the situation.

**Comment by mateusjunges** [comment]
Hello @thetomcake, first of all, thanks for your feedback on this library.
The topic name on the message is used only if you are using the `Kafka::fake()` method, to assert the a message was published specifically on the given topic. I agree with you and already fixed it to be passed to the message automatically, so you don't need to repeat the argument just to use it on tests.
About the documentation, i checked it and in fact is is incorrect. Will be fixed as soon as i release the next version (later today or maximun tomorrow morning). Also, the broker parameter on `publishOn` will be optional, since we can use config files for that.

**Comment by thetomcake** [comment]
> Hello @thetomcake, first of all, thanks for your feedback on this library. The topic name on the message is used only if you are using the `Kafka::fake()` method, to assert the a message was published specifically on the given topic. I agree with you and already fixed it to be passed to the message automatically, so you don't need to repeat the argument just to use it on tests. About the documentation, i checked it and in fact is is incorrect. Will be fixed as soon as i release the next version (later today or maximun tomorrow morning). Also, the broker parameter on `publishOn` will be optional, since we can use config files for that.

No problem @mateusjunges, we're fairly new to Kafka over here, so wanted to make sure I didn't get the complete wrong end of the stick. Glad we could help, we will be using the library in some fairly large apps in the next few weeks so might throw some more PRs your way.

You've answered my issue above, thanks a lot.

**Comment by mateusjunges** [comment]
Hello @thetomcake, i just released [v1.3.0](https://github.com/mateusjunges/laravel-kafka/tree/v1.3.0) of this package. Here's the changelog: https://github.com/mateusjunges/laravel-kafka/blob/v1.3.0/CHANGELOG.md
